File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.rb
# @param {Integer[]} arr
# @return {Boolean}
def unique_occurrences(arr)
    # hash -> count occurrence // O(n)
    # occurrence -> into set  // O(n)
    # compare hash size and occurrence set //O (1)
    count = Hash.new(0)
    arr.each do |element|
        count[element] += 1
    end

    occurences = count.values.to_set

    count.size == occurences.size
end

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
import java.util.*;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int a : arr) {
            int count = map.getOrDefault(a, 0);
            map.put(a, count + 1);
        }
        // if size of keySet and Set of values equal
        Set<Integer> keySet = map.keySet();
        Set<Integer> valueSet = new HashSet<>(map.values());
        return keySet.size() == valueSet.size();
    }
}